ballista/rust/scheduler/src/state/persistent_state.rs

```diff
@@ -124,28 +124,33 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan>
             .get_from_prefix(&get_stage_prefix(&self.namespace))
             .await?;
 
+        let mut tmp_stages: HashMap<StageKey, Arc<dyn ExecutionPlan>> = HashMap::new();
```

Comment:
This makes sense to me (make the snapshot while not holding the lock) but I am not familiar enough with the ballista codebase to know if it would cause problems. Perhaps @mingmwang @yahoNanJing  or @thinkharderdev  could review it too?

Reply:
I don't think so. This should only happen when the scheduler initializes and needs to load the persistent state into memory

Reply:
LGTM